docs/debugging-testing/testing/unit_testing.md

```diff
@@ -61,19 +61,21 @@ In a terminal window:
 
 ## Test code structure
 
-Find unit tests in the Mbed OS repository under the `UNITTESTS` folder of each library. We recommend unit test files use an identical directory path as the file under test. This makes it easier to find unit tests for a particular class or a module. For example, if the file you're testing is `some/example/path/ClassName.cpp`, then all the test files are in the `UNITTESTS/some/example/path/ClassName` directory. Each test suite needs to have its own `unittest.cmake` file for test configuration.
+Find unit tests in the Mbed OS repository under the `UNITTESTS` folder of each library. We recommend unit test files use an identical directory path as the file under test. This makes it easier to find unit tests for a particular class or a module. For example, if the file you're testing is `some/example/path/ClassName.cpp`, then all the test files are in the `UNITTESTS/some/example/path/ClassName` directory. Each test suite needs to have its own `CMakeLists.txt` file for test CMake configuration.
 
-All the class stubs should be located in the Mbed OS root directory `UNITTESTS/stubs`. Multiple test suites can use a single stub class, which should follow the naming convention `ClassName_stub.cpp` for the source file and `ClassName_stub.h` for the header file. Use the actual header files for the unit tests, and don't stub headers if possible. The stubbed headers reside in the `UNITTESTS/target_h` directory.
+All the stub sources are grouped into stub libraries `mbed-stubs-rtos`, `mbed-stubs-drivers`, `mbed-stubs-events`, `mbed-stubs-hal`, `mbed-stubs-platform`, `mbed-stubs-storage`, `mbed-stubs-connectivity` and all stub libraries are target linked under `mbed-stubs`. The library unit test suite CMake is expected to link with the required stub library or directly link with mbed-stubs depends on their test cases.
 
-### Test discovery
+The new stub file should follow the naming convention `ClassName_stub.cpp` for the source file and `ClassName_stub.h` for the header file and add it under respective existing stub library or create their stubs Libary depends on their test cases.
```

Comment:
```suggestion
The new stub file should follow the naming convention `ClassName_stub.cpp` for the source file and `ClassName_stub.h` for the header file. They should be added under their respective existing stub CMake library or create their stubs library depends on their test cases.
```

`...depends on their test cases` , can you give examples when to do one or the other?

Reply:
In case of the file under test calls multiple external APIs (Not part of Mbed OS)  It requires to create a new stub library  to provide that external APIs dummy implementation

Reply:
Please add your explanation in the readme.